libensemble/executors/balsam_executor.py

```diff
@@ -130,32 +179,41 @@ def poll(self):
         balsam_state = self.process.state
         self.runtime = self._get_time_since_balsam_submit()
 
-        if balsam_state in models.END_STATES:
+        if balsam_state in ["RUN_DONE", "POSTPROCESSED", "STAGED_OUT", "JOB_FINISHED"]:
             self._set_complete()
 
-        elif balsam_state in models.ACTIVE_STATES:
-            self.state = 'RUNNING'
+        elif balsam_state in ["RUNNING"]:
+            self.state = "RUNNING"
             self.workdir = self.workdir or self.process.working_directory
 
-        elif (balsam_state in models.PROCESSABLE_STATES or
-              balsam_state in models.RUNNABLE_STATES):
-            self.state = 'WAITING'
+        elif balsam_state in [
+            "CREATED",
+            "AWAITING_PARENTS",
+            "READY",
+            "STAGED_IN",
+            "PREPROCESSED",
+        ]:
+            self.state = "WAITING"
+
+        elif balsam_state in ["RUN_ERROR", "RUN_TIMEOUT", "FAILED"]:
+            self.state = "FAILED"
 
         else:
             raise ExecutorException(
                 "Task state returned from Balsam is not in known list of "
-                "Balsam states. Task state is {}".format(balsam_state))
+                "Balsam states. Task state is {}".format(balsam_state)
+            )
 
     def wait(self, timeout=None):
-        """Waits on completion of the task or raises TimeoutExpired exception
+        """Waits on completion of the task or raises ``TimeoutExpired``.
 
         Status attributes of task are updated on completion.
 
         Parameters
         ----------
 
-        timeout:
-            Time in seconds after which a TimeoutExpired exception is raised"""
+        timeout: int
```

Comment:
Only an `int`, not a `float`?

Reply:
I think ``int`` matches the other executors' APIs.

Plus, less than a second is way too quick to wait on a task that's submitted to, and then scheduled by Balsam across the internet. More than one second might as well be whole seconds

Reply:
Is there a check or issue if a non-int value is given?

Reply:
Or are such quantities cast to ints?